src/process_utility.c

```diff
@@ -2470,6 +2471,30 @@ process_altertable_end_index(Node *parsetree, CollectedCommand *cmd)
 	ts_cache_release(hcache);
 }
 
+static inline void
+process_altertable_chunk_set_tablespace(AlterTableCmd *cmd, Oid relid)
+{
+	Chunk *chunk = ts_chunk_get_by_relid(relid, false);
+
+	if (chunk == NULL)
+		return;
+
+	if (ts_chunk_contains_compressed_data(chunk))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("changing tablespace of compressed chunk is not supported"),
+				 errhint("Please use the corresponding chunk on the uncompressed hypertable "
+						 "instead.")));
+
+	/* set tablespace for compressed chunk */
+	if (chunk->fd.compressed_chunk_id != INVALID_CHUNK_ID)
+	{
+		Chunk *compressed_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
+
+		AlterTableInternal(compressed_chunk->table_id, list_make1(cmd), false);
+	}
```

Comment:
If the chunk is not compressed, shouldn't we set the tablespace for the uncompressed chunk?

Reply:
It will be set automatically here by standard process utility hook call later in `timescaledb_ddl_command_start()` function. Basically it will first alter the compressed chunk here